src/bb-components-build.ts

```diff
@@ -144,13 +146,23 @@ const readtsPrefabs: () => Promise<Prefab[]> = async (): Promise<Prefab[]> => {
     throw new Error(chalk.red('\nPrefabs folder not found\n'));
   }
 
-  const prefabTsFiles: string[] = await readFilesByType(srcDir, 'ts');
-  const prefabTsxFiles: string[] = await readFilesByType(srcDir, 'tsx');
+  const prefabTsFiles: string[] = !isPartial
+    ? await readFilesByType(srcDir, 'ts')
+    : [];
+  const prefabTsxFiles: string[] = !isPartial
+    ? await readFilesByType(srcDir, 'tsx')
+    : [];
+
+  const partialTsxFiles: string[] = isPartial
+    ? await readFilesByType(`${srcDir}/partials`, 'tsx')
+    : [];
```

Comment:
```typescript
let prefabTsFiles: string[] = [];
let prefabTsxFiles: string[] = [];
let partialTsxFiles: string[] = [];

if (isPartial) {
  partialTsxFiles = await readFilesByType(`${srcDir}/partials`, 'tsx');
} else {
  prefabTsFiles = await readFilesByType(srcDir, 'ts');
  prefabTsxFiles = await readFilesByType(srcDir, 'tsx');
}
```

Reply:
Ideally you would want to modify the `readFilesByType` util function so it supports a array of strings e.g` ['tsx', 'ts']`. This way you dont need to call the utill function twice and merge it  on line 160

Reply:
I resolved the comment about the ternaries, As for your feedback Menouer, I completely agree but i'm going to fix that in a seperate ticket because it wil touch also the styles and it will take a bit more time to test this